{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generating config files for the second round of sequencing\n",
    "\n",
    "Also going to update the column names to match up with the standard sensor analysis pipeline (from MBESv2)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "fp = '/Volumes/sanchezrivera/samgould/240624San/'\n",
    "k = os.listdir(fp)\n",
    "k = [i for i in k if i[:3]=='D24'] #might need to change depending on file naming\n",
    "\n",
    "folder_name = []\n",
    "f1 = []\n",
    "f2 = []\n",
    "for ii in range(len(k)):\n",
    "    j = os.listdir(f'{fp}/{k[ii]}')\n",
    "    j = sorted([i for i in j if i[-5:]=='fastq'])\n",
    "\n",
    "    #weird error to deal with\n",
    "    #hacky solution\n",
    "    if len(j)==2:\n",
    "        f1.append(f'./{k[ii]}/{j[0]}')\n",
    "        f2.append(f'./{k[ii]}/{j[1]}')\n",
    "    else:\n",
    "        f1.append(f'./{k[ii]}/{j[2]}')\n",
    "        f2.append(f'./{k[ii]}/{j[3]}')\n",
    "    folder_name.append(k[ii] + '_guide_split_BARCODES')\n",
    "\n",
    "config_df = pd.DataFrame(dict(zip(['ArrayTaskID', 'R1_FILE', 'R2_FILE', 'folder_name'], [list(range(1, len(f1)+1)), f1, f2, folder_name])))\n",
    "#config_df.to_csv('sensor_extraction_fastq_split_config_DIEGO_IDR.txt', index=False, sep = ' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ArrayTaskID</th>\n",
       "      <th>R1_FILE</th>\n",
       "      <th>R2_FILE</th>\n",
       "      <th>folder_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>./D24-10888-6775T/240624San_D24-10888_1_sequen...</td>\n",
       "      <td>./D24-10888-6775T/240624San_D24-10888_2_sequen...</td>\n",
       "      <td>D24-10888-6775T_guide_split_BARCODES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>./D24-10889-6775T/240624San_D24-10889_1_sequen...</td>\n",
       "      <td>./D24-10889-6775T/240624San_D24-10889_2_sequen...</td>\n",
       "      <td>D24-10889-6775T_guide_split_BARCODES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>./D24-10890-6775T/240624San_D24-10890_1_sequen...</td>\n",
       "      <td>./D24-10890-6775T/240624San_D24-10890_2_sequen...</td>\n",
       "      <td>D24-10890-6775T_guide_split_BARCODES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>./D24-10891-6775T/240624San_D24-10891_1_sequen...</td>\n",
       "      <td>./D24-10891-6775T/240624San_D24-10891_2_sequen...</td>\n",
       "      <td>D24-10891-6775T_guide_split_BARCODES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>./D24-10892-6775T/240624San_D24-10892_1_sequen...</td>\n",
       "      <td>./D24-10892-6775T/240624San_D24-10892_2_sequen...</td>\n",
       "      <td>D24-10892-6775T_guide_split_BARCODES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>./D24-10893-6775T/240624San_D24-10893_1_sequen...</td>\n",
       "      <td>./D24-10893-6775T/240624San_D24-10893_2_sequen...</td>\n",
       "      <td>D24-10893-6775T_guide_split_BARCODES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ArrayTaskID                                            R1_FILE  \\\n",
       "0            1  ./D24-10888-6775T/240624San_D24-10888_1_sequen...   \n",
       "1            2  ./D24-10889-6775T/240624San_D24-10889_1_sequen...   \n",
       "2            3  ./D24-10890-6775T/240624San_D24-10890_1_sequen...   \n",
       "3            4  ./D24-10891-6775T/240624San_D24-10891_1_sequen...   \n",
       "4            5  ./D24-10892-6775T/240624San_D24-10892_1_sequen...   \n",
       "5            6  ./D24-10893-6775T/240624San_D24-10893_1_sequen...   \n",
       "\n",
       "                                             R2_FILE  \\\n",
       "0  ./D24-10888-6775T/240624San_D24-10888_2_sequen...   \n",
       "1  ./D24-10889-6775T/240624San_D24-10889_2_sequen...   \n",
       "2  ./D24-10890-6775T/240624San_D24-10890_2_sequen...   \n",
       "3  ./D24-10891-6775T/240624San_D24-10891_2_sequen...   \n",
       "4  ./D24-10892-6775T/240624San_D24-10892_2_sequen...   \n",
       "5  ./D24-10893-6775T/240624San_D24-10893_2_sequen...   \n",
       "\n",
       "                            folder_name  \n",
       "0  D24-10888-6775T_guide_split_BARCODES  \n",
       "1  D24-10889-6775T_guide_split_BARCODES  \n",
       "2  D24-10890-6775T_guide_split_BARCODES  \n",
       "3  D24-10891-6775T_guide_split_BARCODES  \n",
       "4  D24-10892-6775T_guide_split_BARCODES  \n",
       "5  D24-10893-6775T_guide_split_BARCODES  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#config_df.to_csv('be_cluster_scripts_2024_7_1/sensor_extraction_fastq_split_config_DIEGO_IDR_v2.txt', index=False, sep = ' ')\n",
    "config_df = config_df[0:6]\n",
    "\n",
    "#config_df.to_csv('sensor_extraction_fastq_split_config_DIEGO_IDR_v3_barcode.txt', index=False, sep = ' ')\n",
    "config_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Updating the dataframe so that columns are correct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'IDR_ABE_library_final.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/5g/xxjcy2g531n02hnyb6d8qdsr0000gn/T/ipykernel_2621/1044791950.py\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'IDR_ABE_library_final.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m    946\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    947\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 948\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    949\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    950\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    609\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    610\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 611\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    612\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    613\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1446\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1447\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandles\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mIOHandles\u001b[0m \u001b[0;34m|\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1448\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1449\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1450\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1703\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0;34m\"b\"\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1704\u001b[0m                     \u001b[0mmode\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34m\"b\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1705\u001b[0;31m             self.handles = get_handle(\n\u001b[0m\u001b[1;32m   1706\u001b[0m                 \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1707\u001b[0m                 \u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/io/common.py\u001b[0m in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    861\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mioargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;34m\"b\"\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mioargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmode\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    862\u001b[0m             \u001b[0;31m# Encoding\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 863\u001b[0;31m             handle = open(\n\u001b[0m\u001b[1;32m    864\u001b[0m                 \u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    865\u001b[0m                 \u001b[0mioargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'IDR_ABE_library_final.csv'"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('IDR_ABE_library_final.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Hugo_Symbol', 'Chromosome', 'Start_Position', 'HGVSc', 'Variant_Index',\n",
       "       'Variant_Classification', 'Polyphen_Prediction', 'Base_Change',\n",
       "       'HGVSp_Short', 'Mut_PAM_Distance', 'PAM_Strand', 'PAM', 'sgRNA',\n",
       "       'sgRNA_w_PAM', 'Target_Sequence_42_RC', 'Edited_sgRNA_w_PAM',\n",
       "       'unique_BC', 'Final_seq', 'Oligo_ID', 'type_of_guide', 'Entrez_Gene_Id',\n",
       "       'Center', 'NCBI_Build', 'End_Position', 'Strand', 'Consequence',\n",
       "       'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1',\n",
       "       'Tumor_Seq_Allele2', 'dbSNP_RS', 'Tumor_Sample_Barcode',\n",
       "       'Matched_Norm_Sample_Barcode', 'Match_Norm_Seq_Allele1',\n",
       "       'Match_Norm_Seq_Allele2', 'Verification_Status', 'Mutation_Status',\n",
       "       't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count', 'HGVSp',\n",
       "       'Transcript_ID', 'RefSeq', 'Protein_position', 'Codons', 'Exon_Number',\n",
       "       'gnomAD_AF', 'gnomAD_AFR_AF', 'gnomAD_AMR_AF', 'gnomAD_ASJ_AF',\n",
       "       'gnomAD_EAS_AF', 'gnomAD_FIN_AF', 'gnomAD_NFE_AF', 'gnomAD_OTH_AF',\n",
       "       'gnomAD_SAS_AF', 'FILTER', 'Polyphen_Score', 'SIFT_Prediction',\n",
       "       'SIFT_Score', 'n_depth', 't_depth', 'Annotation_Status',\n",
       "       'mutationInCis_Flag', 'num_tumors', 'Target_Sequence_50',\n",
       "       'scaffold_BC_add', 'Full_Seq', 'BC_pair',\n",
       "       'edited_Target_Sequence_42_RC', 'Guide_ID'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>Variant_Index</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Polyphen_Prediction</th>\n",
       "      <th>Base_Change</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Mut_PAM_Distance</th>\n",
       "      <th>...</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>Annotation_Status</th>\n",
       "      <th>mutationInCis_Flag</th>\n",
       "      <th>num_tumors</th>\n",
       "      <th>Target_Sequence_50</th>\n",
       "      <th>scaffold_BC_add</th>\n",
       "      <th>Full_Seq</th>\n",
       "      <th>BC_pair</th>\n",
       "      <th>sensor_alt</th>\n",
       "      <th>gRNA_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>9</td>\n",
       "      <td>133755997</td>\n",
       "      <td>ENST00000318560.5:c.1624A&gt;G</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>benign</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>p.R542G</td>\n",
       "      <td>16.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCATCTCAGGCACGTCAGTGGTGTCTCCGTGCTCTGCAGCTC</td>\n",
       "      <td>sg_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>9</td>\n",
       "      <td>133755997</td>\n",
       "      <td>ENST00000318560.5:c.1624A&gt;G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>benign</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>p.R542G</td>\n",
       "      <td>14.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATCTCAGGCACGTCAGTGGTGTCTCCGTGCTCTGCAGCTCTC</td>\n",
       "      <td>sg_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>9</td>\n",
       "      <td>133760635</td>\n",
       "      <td>ENST00000318560.5:c.2958A&gt;G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>p.P986=</td>\n",
       "      <td>17.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGTCCCCTGCCAGGGCCGAGGATGCTGACGGCAACGTGGAGG</td>\n",
       "      <td>sg_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>1</td>\n",
       "      <td>179078007</td>\n",
       "      <td>ENST00000502732.1:c.2395A&gt;G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>benign</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>p.M799V</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCTGGCAGTTCCTGGGAAGGGTCACTGCCATCCTATCCTGCT</td>\n",
       "      <td>sg_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFF1</td>\n",
       "      <td>4</td>\n",
       "      <td>88035564</td>\n",
       "      <td>ENST00000307808.6:c.1558A&gt;G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>benign</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>p.T520A</td>\n",
       "      <td>13.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCGCAGCTGGCTGGCTGACTTTGGCCAGCCAGTTGTCCAGCT</td>\n",
       "      <td>sg_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1667</th>\n",
       "      <td>controlsgRNA_95</td>\n",
       "      <td>13</td>\n",
       "      <td>88376208</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>-</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTGAGCTATTGTACAGGGCTGGGGCCATGGGAGACAGTGTTGGTGG...</td>\n",
       "      <td>GGGGCCATGGGAGACAGTGTGTTTAAGAGCTATGCTGGAAACAGCA...</td>\n",
       "      <td>CATAGCGTACACGTCTCACACCGGGGCCATGGGAGACAGTGTGTTT...</td>\n",
       "      <td>F1-R4</td>\n",
       "      <td>CCACAGCCACCAACACTGTCTCCCATAGCCCCAGCCCTGTAC</td>\n",
       "      <td>sg_1668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1668</th>\n",
       "      <td>controlsgRNA_96</td>\n",
       "      <td>X</td>\n",
       "      <td>36797961</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>-</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCTGGACTGGTTGCAACCTAGACCCATGCACCTTCTTCATTGGTCT...</td>\n",
       "      <td>GACCCATGCACCTTCTTCATGTTTAAGAGCTATGCTGGAAACAGCA...</td>\n",
       "      <td>CATAGCGTACACGTCTCACACCGACCCATGCACCTTCTTCATGTTT...</td>\n",
       "      <td>F1-R4</td>\n",
       "      <td>TAGGGGAGACCAATGAAGAAGGTGCACGGGTCTAGGTTGCAA</td>\n",
       "      <td>sg_1669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1669</th>\n",
       "      <td>controlsgRNA_97</td>\n",
       "      <td>10</td>\n",
       "      <td>57998002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>-</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGTGTGGCATGACACTGTAGGAAGTCCTCAAAACTAATCAGGGCAG...</td>\n",
       "      <td>GAAGTCCTCAAAACTAATCAGTTTAAGAGCTATGCTGGAAACAGCA...</td>\n",
       "      <td>CATAGCGTACACGTCTCACACCGAAGTCCTCAAAACTAATCAGTTT...</td>\n",
       "      <td>F1-R4</td>\n",
       "      <td>TTTCCCCTGCCCTGATTAGTTTTGAGAACTTCCTACAGTGTC</td>\n",
       "      <td>sg_1670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1670</th>\n",
       "      <td>controlsgRNA_98</td>\n",
       "      <td>4</td>\n",
       "      <td>81701927</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>-</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TATACAAAATATGTAATCCTGTGAGACATTATATATTGACTGGGTA...</td>\n",
       "      <td>GTGAGACATTATATATTGACGTTTAAGAGCTATGCTGGAAACAGCA...</td>\n",
       "      <td>CATAGCGTACACGTCTCACACCGTGAGACATTATATATTGACGTTT...</td>\n",
       "      <td>F1-R4</td>\n",
       "      <td>TCTGTCTACCCAGTCAATATATAATGCCTCACAGGATTACAT</td>\n",
       "      <td>sg_1671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1671</th>\n",
       "      <td>controlsgRNA_99</td>\n",
       "      <td>14</td>\n",
       "      <td>97634996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>-</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAGGGAAGAATCTCATTGCAGAAGCAAGGAAAAGTGACAGTGGTCC...</td>\n",
       "      <td>GAAGCAAGGAAAAGTGACAGGTTTAAGAGCTATGCTGGAAACAGCA...</td>\n",
       "      <td>CATAGCGTACACGTCTCACACCGAAGCAAGGAAAAGTGACAGGTTT...</td>\n",
       "      <td>F1-R4</td>\n",
       "      <td>AGTGTGGGACCACTGTCACTTTTCCTCGCTTCTGCAATGAGA</td>\n",
       "      <td>sg_1672</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1672 rows × 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Hugo_Symbol Chromosome  Start_Position                        HGVSc  \\\n",
       "0                ABL1          9       133755997  ENST00000318560.5:c.1624A>G   \n",
       "1                ABL1          9       133755997  ENST00000318560.5:c.1624A>G   \n",
       "2                ABL1          9       133760635  ENST00000318560.5:c.2958A>G   \n",
       "3                ABL2          1       179078007  ENST00000502732.1:c.2395A>G   \n",
       "4                AFF1          4        88035564  ENST00000307808.6:c.1558A>G   \n",
       "...               ...        ...             ...                          ...   \n",
       "1667  controlsgRNA_95         13        88376208                          NaN   \n",
       "1668  controlsgRNA_96          X        36797961                          NaN   \n",
       "1669  controlsgRNA_97         10        57998002                          NaN   \n",
       "1670  controlsgRNA_98          4        81701927                          NaN   \n",
       "1671  controlsgRNA_99         14        97634996                          NaN   \n",
       "\n",
       "     Variant_Index Variant_Classification Polyphen_Prediction Base_Change  \\\n",
       "0              2.0      Missense_Mutation              benign         A>G   \n",
       "1              1.0      Missense_Mutation              benign         A>G   \n",
       "2              1.0                 Silent                 NaN         A>G   \n",
       "3              1.0      Missense_Mutation              benign         T>C   \n",
       "4              1.0      Missense_Mutation              benign         A>G   \n",
       "...            ...                    ...                 ...         ...   \n",
       "1667             -                    NaN                 NaN         C>T   \n",
       "1668             -                    NaN                 NaN         A>G   \n",
       "1669             -                    NaN                 NaN         C>T   \n",
       "1670             -                    NaN                 NaN         A>G   \n",
       "1671             -                    NaN                 NaN         A>G   \n",
       "\n",
       "     HGVSp_Short  Mut_PAM_Distance  ... t_depth Annotation_Status  \\\n",
       "0        p.R542G              16.0  ...     NaN               NaN   \n",
       "1        p.R542G              14.0  ...     NaN               NaN   \n",
       "2        p.P986=              17.0  ...     NaN               NaN   \n",
       "3        p.M799V             -13.0  ...     NaN               NaN   \n",
       "4        p.T520A              13.0  ...     NaN               NaN   \n",
       "...          ...               ...  ...     ...               ...   \n",
       "1667           -              15.0  ...     NaN               NaN   \n",
       "1668           -              15.0  ...     NaN               NaN   \n",
       "1669           -              15.0  ...     NaN               NaN   \n",
       "1670           -              15.0  ...     NaN               NaN   \n",
       "1671           -              15.0  ...     NaN               NaN   \n",
       "\n",
       "     mutationInCis_Flag num_tumors  \\\n",
       "0                   NaN        NaN   \n",
       "1                   NaN        NaN   \n",
       "2                   NaN        NaN   \n",
       "3                   NaN        NaN   \n",
       "4                   NaN        NaN   \n",
       "...                 ...        ...   \n",
       "1667                NaN        NaN   \n",
       "1668                NaN        NaN   \n",
       "1669                NaN        NaN   \n",
       "1670                NaN        NaN   \n",
       "1671                NaN        NaN   \n",
       "\n",
       "                                     Target_Sequence_50  \\\n",
       "0                                                   NaN   \n",
       "1                                                   NaN   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4                                                   NaN   \n",
       "...                                                 ...   \n",
       "1667  TTGAGCTATTGTACAGGGCTGGGGCCATGGGAGACAGTGTTGGTGG...   \n",
       "1668  CCTGGACTGGTTGCAACCTAGACCCATGCACCTTCTTCATTGGTCT...   \n",
       "1669  GGTGTGGCATGACACTGTAGGAAGTCCTCAAAACTAATCAGGGCAG...   \n",
       "1670  TATACAAAATATGTAATCCTGTGAGACATTATATATTGACTGGGTA...   \n",
       "1671  AAGGGAAGAATCTCATTGCAGAAGCAAGGAAAAGTGACAGTGGTCC...   \n",
       "\n",
       "                                        scaffold_BC_add  \\\n",
       "0                                                   NaN   \n",
       "1                                                   NaN   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4                                                   NaN   \n",
       "...                                                 ...   \n",
       "1667  GGGGCCATGGGAGACAGTGTGTTTAAGAGCTATGCTGGAAACAGCA...   \n",
       "1668  GACCCATGCACCTTCTTCATGTTTAAGAGCTATGCTGGAAACAGCA...   \n",
       "1669  GAAGTCCTCAAAACTAATCAGTTTAAGAGCTATGCTGGAAACAGCA...   \n",
       "1670  GTGAGACATTATATATTGACGTTTAAGAGCTATGCTGGAAACAGCA...   \n",
       "1671  GAAGCAAGGAAAAGTGACAGGTTTAAGAGCTATGCTGGAAACAGCA...   \n",
       "\n",
       "                                               Full_Seq BC_pair  \\\n",
       "0                                                   NaN     NaN   \n",
       "1                                                   NaN     NaN   \n",
       "2                                                   NaN     NaN   \n",
       "3                                                   NaN     NaN   \n",
       "4                                                   NaN     NaN   \n",
       "...                                                 ...     ...   \n",
       "1667  CATAGCGTACACGTCTCACACCGGGGCCATGGGAGACAGTGTGTTT...   F1-R4   \n",
       "1668  CATAGCGTACACGTCTCACACCGACCCATGCACCTTCTTCATGTTT...   F1-R4   \n",
       "1669  CATAGCGTACACGTCTCACACCGAAGTCCTCAAAACTAATCAGTTT...   F1-R4   \n",
       "1670  CATAGCGTACACGTCTCACACCGTGAGACATTATATATTGACGTTT...   F1-R4   \n",
       "1671  CATAGCGTACACGTCTCACACCGAAGCAAGGAAAAGTGACAGGTTT...   F1-R4   \n",
       "\n",
       "                                      sensor_alt  gRNA_id  \n",
       "0     GCATCTCAGGCACGTCAGTGGTGTCTCCGTGCTCTGCAGCTC     sg_1  \n",
       "1     ATCTCAGGCACGTCAGTGGTGTCTCCGTGCTCTGCAGCTCTC     sg_2  \n",
       "2     GGTCCCCTGCCAGGGCCGAGGATGCTGACGGCAACGTGGAGG     sg_3  \n",
       "3     TCTGGCAGTTCCTGGGAAGGGTCACTGCCATCCTATCCTGCT     sg_4  \n",
       "4     GCGCAGCTGGCTGGCTGACTTTGGCCAGCCAGTTGTCCAGCT     sg_5  \n",
       "...                                          ...      ...  \n",
       "1667  CCACAGCCACCAACACTGTCTCCCATAGCCCCAGCCCTGTAC  sg_1668  \n",
       "1668  TAGGGGAGACCAATGAAGAAGGTGCACGGGTCTAGGTTGCAA  sg_1669  \n",
       "1669  TTTCCCCTGCCCTGATTAGTTTTGAGAACTTCCTACAGTGTC  sg_1670  \n",
       "1670  TCTGTCTACCCAGTCAATATATAATGCCTCACAGGATTACAT  sg_1671  \n",
       "1671  AGTGTGGGACCACTGTCACTTTTCCTCGCTTCTGCAATGAGA  sg_1672  \n",
       "\n",
       "[1672 rows x 71 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_rename = {'Guide_ID':'gRNA_id', 'unique_BC':'Hamming_BC', 'sgRNA':'Protospacer', 'edited_Target_Sequence_42_RC': 'sensor_alt', 'Target_Sequence_42_RC': 'sensor_wt'}\n",
    "\n",
    "df2 = df.rename(columns = col_rename)\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['sg_1.fastq',\n",
       " 'sg_2.fastq',\n",
       " 'sg_3.fastq',\n",
       " 'sg_4.fastq',\n",
       " 'sg_5.fastq',\n",
       " 'sg_6.fastq',\n",
       " 'sg_7.fastq',\n",
       " 'sg_8.fastq',\n",
       " 'sg_9.fastq',\n",
       " 'sg_10.fastq',\n",
       " 'sg_11.fastq',\n",
       " 'sg_12.fastq',\n",
       " 'sg_13.fastq',\n",
       " 'sg_14.fastq',\n",
       " 'sg_15.fastq',\n",
       " 'sg_16.fastq',\n",
       " 'sg_17.fastq',\n",
       " 'sg_18.fastq',\n",
       " 'sg_19.fastq',\n",
       " 'sg_20.fastq',\n",
       " 'sg_21.fastq',\n",
       " 'sg_22.fastq',\n",
       " 'sg_23.fastq',\n",
       " 'sg_24.fastq',\n",
       " 'sg_25.fastq',\n",
       " 'sg_26.fastq',\n",
       " 'sg_27.fastq',\n",
       " 'sg_28.fastq',\n",
       " 'sg_29.fastq',\n",
       " 'sg_30.fastq',\n",
       " 'sg_31.fastq',\n",
       " 'sg_32.fastq',\n",
       " 'sg_33.fastq',\n",
       " 'sg_34.fastq',\n",
       " 'sg_35.fastq',\n",
       " 'sg_36.fastq',\n",
       " 'sg_37.fastq',\n",
       " 'sg_38.fastq',\n",
       " 'sg_39.fastq',\n",
       " 'sg_40.fastq',\n",
       " 'sg_41.fastq',\n",
       " 'sg_42.fastq',\n",
       " 'sg_43.fastq',\n",
       " 'sg_44.fastq',\n",
       " 'sg_45.fastq',\n",
       " 'sg_46.fastq',\n",
       " 'sg_47.fastq',\n",
       " 'sg_48.fastq',\n",
       " 'sg_49.fastq',\n",
       " 'sg_50.fastq',\n",
       " 'sg_51.fastq',\n",
       " 'sg_52.fastq',\n",
       " 'sg_53.fastq',\n",
       " 'sg_54.fastq',\n",
       " 'sg_55.fastq',\n",
       " 'sg_56.fastq',\n",
       " 'sg_57.fastq',\n",
       " 'sg_58.fastq',\n",
       " 'sg_59.fastq',\n",
       " 'sg_60.fastq',\n",
       " 'sg_61.fastq',\n",
       " 'sg_62.fastq',\n",
       " 'sg_63.fastq',\n",
       " 'sg_64.fastq',\n",
       " 'sg_65.fastq',\n",
       " 'sg_66.fastq',\n",
       " 'sg_67.fastq',\n",
       " 'sg_68.fastq',\n",
       " 'sg_69.fastq',\n",
       " 'sg_70.fastq',\n",
       " 'sg_71.fastq',\n",
       " 'sg_72.fastq',\n",
       " 'sg_73.fastq',\n",
       " 'sg_74.fastq',\n",
       " 'sg_75.fastq',\n",
       " 'sg_76.fastq',\n",
       " 'sg_77.fastq',\n",
       " 'sg_78.fastq',\n",
       " 'sg_79.fastq',\n",
       " 'sg_80.fastq',\n",
       " 'sg_81.fastq',\n",
       " 'sg_82.fastq',\n",
       " 'sg_83.fastq',\n",
       " 'sg_84.fastq',\n",
       " 'sg_85.fastq',\n",
       " 'sg_86.fastq',\n",
       " 'sg_87.fastq',\n",
       " 'sg_88.fastq',\n",
       " 'sg_89.fastq',\n",
       " 'sg_90.fastq',\n",
       " 'sg_91.fastq',\n",
       " 'sg_92.fastq',\n",
       " 'sg_93.fastq',\n",
       " 'sg_94.fastq',\n",
       " 'sg_95.fastq',\n",
       " 'sg_96.fastq',\n",
       " 'sg_97.fastq',\n",
       " 'sg_98.fastq',\n",
       " 'sg_99.fastq',\n",
       " 'sg_100.fastq',\n",
       " 'sg_101.fastq',\n",
       " 'sg_102.fastq',\n",
       " 'sg_103.fastq',\n",
       " 'sg_104.fastq',\n",
       " 'sg_105.fastq',\n",
       " 'sg_106.fastq',\n",
       " 'sg_107.fastq',\n",
       " 'sg_108.fastq',\n",
       " 'sg_109.fastq',\n",
       " 'sg_110.fastq',\n",
       " 'sg_111.fastq',\n",
       " 'sg_112.fastq',\n",
       " 'sg_113.fastq',\n",
       " 'sg_114.fastq',\n",
       " 'sg_115.fastq',\n",
       " 'sg_116.fastq',\n",
       " 'sg_117.fastq',\n",
       " 'sg_118.fastq',\n",
       " 'sg_119.fastq',\n",
       " 'sg_120.fastq',\n",
       " 'sg_121.fastq',\n",
       " 'sg_122.fastq',\n",
       " 'sg_123.fastq',\n",
       " 'sg_124.fastq',\n",
       " 'sg_125.fastq',\n",
       " 'sg_126.fastq',\n",
       " 'sg_127.fastq',\n",
       " 'sg_128.fastq',\n",
       " 'sg_129.fastq',\n",
       " 'sg_130.fastq',\n",
       " 'sg_131.fastq',\n",
       " 'sg_132.fastq',\n",
       " 'sg_133.fastq',\n",
       " 'sg_134.fastq',\n",
       " 'sg_135.fastq',\n",
       " 'sg_136.fastq',\n",
       " 'sg_137.fastq',\n",
       " 'sg_138.fastq',\n",
       " 'sg_139.fastq',\n",
       " 'sg_140.fastq',\n",
       " 'sg_141.fastq',\n",
       " 'sg_142.fastq',\n",
       " 'sg_143.fastq',\n",
       " 'sg_144.fastq',\n",
       " 'sg_145.fastq',\n",
       " 'sg_146.fastq',\n",
       " 'sg_147.fastq',\n",
       " 'sg_148.fastq',\n",
       " 'sg_149.fastq',\n",
       " 'sg_150.fastq',\n",
       " 'sg_151.fastq',\n",
       " 'sg_152.fastq',\n",
       " 'sg_153.fastq',\n",
       " 'sg_154.fastq',\n",
       " 'sg_155.fastq',\n",
       " 'sg_156.fastq',\n",
       " 'sg_157.fastq',\n",
       " 'sg_158.fastq',\n",
       " 'sg_159.fastq',\n",
       " 'sg_160.fastq',\n",
       " 'sg_161.fastq',\n",
       " 'sg_162.fastq',\n",
       " 'sg_163.fastq',\n",
       " 'sg_164.fastq',\n",
       " 'sg_165.fastq',\n",
       " 'sg_166.fastq',\n",
       " 'sg_167.fastq',\n",
       " 'sg_168.fastq',\n",
       " 'sg_169.fastq',\n",
       " 'sg_170.fastq',\n",
       " 'sg_171.fastq',\n",
       " 'sg_172.fastq',\n",
       " 'sg_173.fastq',\n",
       " 'sg_174.fastq',\n",
       " 'sg_175.fastq',\n",
       " 'sg_176.fastq',\n",
       " 'sg_177.fastq',\n",
       " 'sg_178.fastq',\n",
       " 'sg_179.fastq',\n",
       " 'sg_180.fastq',\n",
       " 'sg_181.fastq',\n",
       " 'sg_182.fastq',\n",
       " 'sg_183.fastq',\n",
       " 'sg_184.fastq',\n",
       " 'sg_185.fastq',\n",
       " 'sg_186.fastq',\n",
       " 'sg_187.fastq',\n",
       " 'sg_188.fastq',\n",
       " 'sg_189.fastq',\n",
       " 'sg_190.fastq',\n",
       " 'sg_191.fastq',\n",
       " 'sg_192.fastq',\n",
       " 'sg_193.fastq',\n",
       " 'sg_194.fastq',\n",
       " 'sg_195.fastq',\n",
       " 'sg_196.fastq',\n",
       " 'sg_197.fastq',\n",
       " 'sg_198.fastq',\n",
       " 'sg_199.fastq',\n",
       " 'sg_200.fastq',\n",
       " 'sg_201.fastq',\n",
       " 'sg_202.fastq',\n",
       " 'sg_203.fastq',\n",
       " 'sg_204.fastq',\n",
       " 'sg_205.fastq',\n",
       " 'sg_206.fastq',\n",
       " 'sg_207.fastq',\n",
       " 'sg_208.fastq',\n",
       " 'sg_209.fastq',\n",
       " 'sg_210.fastq',\n",
       " 'sg_211.fastq',\n",
       " 'sg_212.fastq',\n",
       " 'sg_213.fastq',\n",
       " 'sg_214.fastq',\n",
       " 'sg_215.fastq',\n",
       " 'sg_216.fastq',\n",
       " 'sg_217.fastq',\n",
       " 'sg_218.fastq',\n",
       " 'sg_219.fastq',\n",
       " 'sg_220.fastq',\n",
       " 'sg_221.fastq',\n",
       " 'sg_222.fastq',\n",
       " 'sg_223.fastq',\n",
       " 'sg_224.fastq',\n",
       " 'sg_225.fastq',\n",
       " 'sg_226.fastq',\n",
       " 'sg_227.fastq',\n",
       " 'sg_228.fastq',\n",
       " 'sg_229.fastq',\n",
       " 'sg_230.fastq',\n",
       " 'sg_231.fastq',\n",
       " 'sg_232.fastq',\n",
       " 'sg_233.fastq',\n",
       " 'sg_234.fastq',\n",
       " 'sg_235.fastq',\n",
       " 'sg_236.fastq',\n",
       " 'sg_237.fastq',\n",
       " 'sg_238.fastq',\n",
       " 'sg_239.fastq',\n",
       " 'sg_240.fastq',\n",
       " 'sg_241.fastq',\n",
       " 'sg_242.fastq',\n",
       " 'sg_243.fastq',\n",
       " 'sg_244.fastq',\n",
       " 'sg_245.fastq',\n",
       " 'sg_246.fastq',\n",
       " 'sg_247.fastq',\n",
       " 'sg_248.fastq',\n",
       " 'sg_249.fastq',\n",
       " 'sg_250.fastq',\n",
       " 'sg_251.fastq',\n",
       " 'sg_252.fastq',\n",
       " 'sg_253.fastq',\n",
       " 'sg_254.fastq',\n",
       " 'sg_255.fastq',\n",
       " 'sg_256.fastq',\n",
       " 'sg_257.fastq',\n",
       " 'sg_258.fastq',\n",
       " 'sg_259.fastq',\n",
       " 'sg_260.fastq',\n",
       " 'sg_261.fastq',\n",
       " 'sg_262.fastq',\n",
       " 'sg_263.fastq',\n",
       " 'sg_264.fastq',\n",
       " 'sg_265.fastq',\n",
       " 'sg_266.fastq',\n",
       " 'sg_267.fastq',\n",
       " 'sg_268.fastq',\n",
       " 'sg_269.fastq',\n",
       " 'sg_270.fastq',\n",
       " 'sg_271.fastq',\n",
       " 'sg_272.fastq',\n",
       " 'sg_273.fastq',\n",
       " 'sg_274.fastq',\n",
       " 'sg_275.fastq',\n",
       " 'sg_276.fastq',\n",
       " 'sg_277.fastq',\n",
       " 'sg_278.fastq',\n",
       " 'sg_279.fastq',\n",
       " 'sg_280.fastq',\n",
       " 'sg_281.fastq',\n",
       " 'sg_282.fastq',\n",
       " 'sg_283.fastq',\n",
       " 'sg_284.fastq',\n",
       " 'sg_285.fastq',\n",
       " 'sg_286.fastq',\n",
       " 'sg_287.fastq',\n",
       " 'sg_288.fastq',\n",
       " 'sg_289.fastq',\n",
       " 'sg_290.fastq',\n",
       " 'sg_291.fastq',\n",
       " 'sg_292.fastq',\n",
       " 'sg_293.fastq',\n",
       " 'sg_294.fastq',\n",
       " 'sg_295.fastq',\n",
       " 'sg_296.fastq',\n",
       " 'sg_297.fastq',\n",
       " 'sg_298.fastq',\n",
       " 'sg_299.fastq',\n",
       " 'sg_300.fastq',\n",
       " 'sg_301.fastq',\n",
       " 'sg_302.fastq',\n",
       " 'sg_303.fastq',\n",
       " 'sg_304.fastq',\n",
       " 'sg_305.fastq',\n",
       " 'sg_306.fastq',\n",
       " 'sg_307.fastq',\n",
       " 'sg_308.fastq',\n",
       " 'sg_309.fastq',\n",
       " 'sg_310.fastq',\n",
       " 'sg_311.fastq',\n",
       " 'sg_312.fastq',\n",
       " 'sg_313.fastq',\n",
       " 'sg_314.fastq',\n",
       " 'sg_315.fastq',\n",
       " 'sg_316.fastq',\n",
       " 'sg_317.fastq',\n",
       " 'sg_318.fastq',\n",
       " 'sg_319.fastq',\n",
       " 'sg_320.fastq',\n",
       " 'sg_321.fastq',\n",
       " 'sg_322.fastq',\n",
       " 'sg_323.fastq',\n",
       " 'sg_324.fastq',\n",
       " 'sg_325.fastq',\n",
       " 'sg_326.fastq',\n",
       " 'sg_327.fastq',\n",
       " 'sg_328.fastq',\n",
       " 'sg_329.fastq',\n",
       " 'sg_330.fastq',\n",
       " 'sg_331.fastq',\n",
       " 'sg_332.fastq',\n",
       " 'sg_333.fastq',\n",
       " 'sg_334.fastq',\n",
       " 'sg_335.fastq',\n",
       " 'sg_336.fastq',\n",
       " 'sg_337.fastq',\n",
       " 'sg_338.fastq',\n",
       " 'sg_339.fastq',\n",
       " 'sg_340.fastq',\n",
       " 'sg_341.fastq',\n",
       " 'sg_342.fastq',\n",
       " 'sg_343.fastq',\n",
       " 'sg_344.fastq',\n",
       " 'sg_345.fastq',\n",
       " 'sg_346.fastq',\n",
       " 'sg_347.fastq',\n",
       " 'sg_348.fastq',\n",
       " 'sg_349.fastq',\n",
       " 'sg_350.fastq',\n",
       " 'sg_351.fastq',\n",
       " 'sg_352.fastq',\n",
       " 'sg_353.fastq',\n",
       " 'sg_354.fastq',\n",
       " 'sg_355.fastq',\n",
       " 'sg_356.fastq',\n",
       " 'sg_357.fastq',\n",
       " 'sg_358.fastq',\n",
       " 'sg_359.fastq',\n",
       " 'sg_360.fastq',\n",
       " 'sg_361.fastq',\n",
       " 'sg_362.fastq',\n",
       " 'sg_363.fastq',\n",
       " 'sg_364.fastq',\n",
       " 'sg_365.fastq',\n",
       " 'sg_366.fastq',\n",
       " 'sg_367.fastq',\n",
       " 'sg_368.fastq',\n",
       " 'sg_369.fastq',\n",
       " 'sg_370.fastq',\n",
       " 'sg_371.fastq',\n",
       " 'sg_372.fastq',\n",
       " 'sg_373.fastq',\n",
       " 'sg_374.fastq',\n",
       " 'sg_375.fastq',\n",
       " 'sg_376.fastq',\n",
       " 'sg_377.fastq',\n",
       " 'sg_378.fastq',\n",
       " 'sg_379.fastq',\n",
       " 'sg_380.fastq',\n",
       " 'sg_381.fastq',\n",
       " 'sg_382.fastq',\n",
       " 'sg_383.fastq',\n",
       " 'sg_384.fastq',\n",
       " 'sg_385.fastq',\n",
       " 'sg_386.fastq',\n",
       " 'sg_387.fastq',\n",
       " 'sg_388.fastq',\n",
       " 'sg_389.fastq',\n",
       " 'sg_390.fastq',\n",
       " 'sg_391.fastq',\n",
       " 'sg_392.fastq',\n",
       " 'sg_393.fastq',\n",
       " 'sg_394.fastq',\n",
       " 'sg_395.fastq',\n",
       " 'sg_396.fastq',\n",
       " 'sg_397.fastq',\n",
       " 'sg_398.fastq',\n",
       " 'sg_399.fastq',\n",
       " 'sg_400.fastq',\n",
       " 'sg_401.fastq',\n",
       " 'sg_402.fastq',\n",
       " 'sg_403.fastq',\n",
       " 'sg_404.fastq',\n",
       " 'sg_405.fastq',\n",
       " 'sg_406.fastq',\n",
       " 'sg_407.fastq',\n",
       " 'sg_408.fastq',\n",
       " 'sg_409.fastq',\n",
       " 'sg_410.fastq',\n",
       " 'sg_411.fastq',\n",
       " 'sg_412.fastq',\n",
       " 'sg_413.fastq',\n",
       " 'sg_414.fastq',\n",
       " 'sg_415.fastq',\n",
       " 'sg_416.fastq',\n",
       " 'sg_417.fastq',\n",
       " 'sg_418.fastq',\n",
       " 'sg_419.fastq',\n",
       " 'sg_420.fastq',\n",
       " 'sg_421.fastq',\n",
       " 'sg_422.fastq',\n",
       " 'sg_423.fastq',\n",
       " 'sg_424.fastq',\n",
       " 'sg_425.fastq',\n",
       " 'sg_426.fastq',\n",
       " 'sg_427.fastq',\n",
       " 'sg_428.fastq',\n",
       " 'sg_429.fastq',\n",
       " 'sg_430.fastq',\n",
       " 'sg_431.fastq',\n",
       " 'sg_432.fastq',\n",
       " 'sg_433.fastq',\n",
       " 'sg_434.fastq',\n",
       " 'sg_435.fastq',\n",
       " 'sg_436.fastq',\n",
       " 'sg_437.fastq',\n",
       " 'sg_438.fastq',\n",
       " 'sg_439.fastq',\n",
       " 'sg_440.fastq',\n",
       " 'sg_441.fastq',\n",
       " 'sg_442.fastq',\n",
       " 'sg_443.fastq',\n",
       " 'sg_444.fastq',\n",
       " 'sg_445.fastq',\n",
       " 'sg_446.fastq',\n",
       " 'sg_447.fastq',\n",
       " 'sg_448.fastq',\n",
       " 'sg_449.fastq',\n",
       " 'sg_450.fastq',\n",
       " 'sg_451.fastq',\n",
       " 'sg_452.fastq',\n",
       " 'sg_453.fastq',\n",
       " 'sg_454.fastq',\n",
       " 'sg_455.fastq',\n",
       " 'sg_456.fastq',\n",
       " 'sg_457.fastq',\n",
       " 'sg_458.fastq',\n",
       " 'sg_459.fastq',\n",
       " 'sg_460.fastq',\n",
       " 'sg_461.fastq',\n",
       " 'sg_462.fastq',\n",
       " 'sg_463.fastq',\n",
       " 'sg_464.fastq',\n",
       " 'sg_465.fastq',\n",
       " 'sg_466.fastq',\n",
       " 'sg_467.fastq',\n",
       " 'sg_468.fastq',\n",
       " 'sg_469.fastq',\n",
       " 'sg_470.fastq',\n",
       " 'sg_471.fastq',\n",
       " 'sg_472.fastq',\n",
       " 'sg_473.fastq',\n",
       " 'sg_474.fastq',\n",
       " 'sg_475.fastq',\n",
       " 'sg_476.fastq',\n",
       " 'sg_477.fastq',\n",
       " 'sg_478.fastq',\n",
       " 'sg_479.fastq',\n",
       " 'sg_480.fastq',\n",
       " 'sg_481.fastq',\n",
       " 'sg_482.fastq',\n",
       " 'sg_483.fastq',\n",
       " 'sg_484.fastq',\n",
       " 'sg_485.fastq',\n",
       " 'sg_486.fastq',\n",
       " 'sg_487.fastq',\n",
       " 'sg_488.fastq',\n",
       " 'sg_489.fastq',\n",
       " 'sg_490.fastq',\n",
       " 'sg_491.fastq',\n",
       " 'sg_492.fastq',\n",
       " 'sg_493.fastq',\n",
       " 'sg_494.fastq',\n",
       " 'sg_495.fastq',\n",
       " 'sg_496.fastq',\n",
       " 'sg_497.fastq',\n",
       " 'sg_498.fastq',\n",
       " 'sg_499.fastq',\n",
       " 'sg_500.fastq',\n",
       " 'sg_501.fastq',\n",
       " 'sg_502.fastq',\n",
       " 'sg_503.fastq',\n",
       " 'sg_504.fastq',\n",
       " 'sg_505.fastq',\n",
       " 'sg_506.fastq',\n",
       " 'sg_507.fastq',\n",
       " 'sg_508.fastq',\n",
       " 'sg_509.fastq',\n",
       " 'sg_510.fastq',\n",
       " 'sg_511.fastq',\n",
       " 'sg_512.fastq',\n",
       " 'sg_513.fastq',\n",
       " 'sg_514.fastq',\n",
       " 'sg_515.fastq',\n",
       " 'sg_516.fastq',\n",
       " 'sg_517.fastq',\n",
       " 'sg_518.fastq',\n",
       " 'sg_519.fastq',\n",
       " 'sg_520.fastq',\n",
       " 'sg_521.fastq',\n",
       " 'sg_522.fastq',\n",
       " 'sg_523.fastq',\n",
       " 'sg_524.fastq',\n",
       " 'sg_525.fastq',\n",
       " 'sg_526.fastq',\n",
       " 'sg_527.fastq',\n",
       " 'sg_528.fastq',\n",
       " 'sg_529.fastq',\n",
       " 'sg_530.fastq',\n",
       " 'sg_531.fastq',\n",
       " 'sg_532.fastq',\n",
       " 'sg_533.fastq',\n",
       " 'sg_534.fastq',\n",
       " 'sg_535.fastq',\n",
       " 'sg_536.fastq',\n",
       " 'sg_537.fastq',\n",
       " 'sg_538.fastq',\n",
       " 'sg_539.fastq',\n",
       " 'sg_540.fastq',\n",
       " 'sg_541.fastq',\n",
       " 'sg_542.fastq',\n",
       " 'sg_543.fastq',\n",
       " 'sg_544.fastq',\n",
       " 'sg_545.fastq',\n",
       " 'sg_546.fastq',\n",
       " 'sg_547.fastq',\n",
       " 'sg_548.fastq',\n",
       " 'sg_549.fastq',\n",
       " 'sg_550.fastq',\n",
       " 'sg_551.fastq',\n",
       " 'sg_552.fastq',\n",
       " 'sg_553.fastq',\n",
       " 'sg_554.fastq',\n",
       " 'sg_555.fastq',\n",
       " 'sg_556.fastq',\n",
       " 'sg_557.fastq',\n",
       " 'sg_558.fastq',\n",
       " 'sg_559.fastq',\n",
       " 'sg_560.fastq',\n",
       " 'sg_561.fastq',\n",
       " 'sg_562.fastq',\n",
       " 'sg_563.fastq',\n",
       " 'sg_564.fastq',\n",
       " 'sg_565.fastq',\n",
       " 'sg_566.fastq',\n",
       " 'sg_567.fastq',\n",
       " 'sg_568.fastq',\n",
       " 'sg_569.fastq',\n",
       " 'sg_570.fastq',\n",
       " 'sg_571.fastq',\n",
       " 'sg_572.fastq',\n",
       " 'sg_573.fastq',\n",
       " 'sg_574.fastq',\n",
       " 'sg_575.fastq',\n",
       " 'sg_576.fastq',\n",
       " 'sg_577.fastq',\n",
       " 'sg_578.fastq',\n",
       " 'sg_579.fastq',\n",
       " 'sg_580.fastq',\n",
       " 'sg_581.fastq',\n",
       " 'sg_582.fastq',\n",
       " 'sg_583.fastq',\n",
       " 'sg_584.fastq',\n",
       " 'sg_585.fastq',\n",
       " 'sg_586.fastq',\n",
       " 'sg_587.fastq',\n",
       " 'sg_588.fastq',\n",
       " 'sg_589.fastq',\n",
       " 'sg_590.fastq',\n",
       " 'sg_591.fastq',\n",
       " 'sg_592.fastq',\n",
       " 'sg_593.fastq',\n",
       " 'sg_594.fastq',\n",
       " 'sg_595.fastq',\n",
       " 'sg_596.fastq',\n",
       " 'sg_597.fastq',\n",
       " 'sg_598.fastq',\n",
       " 'sg_599.fastq',\n",
       " 'sg_600.fastq',\n",
       " 'sg_601.fastq',\n",
       " 'sg_602.fastq',\n",
       " 'sg_603.fastq',\n",
       " 'sg_604.fastq',\n",
       " 'sg_605.fastq',\n",
       " 'sg_606.fastq',\n",
       " 'sg_607.fastq',\n",
       " 'sg_608.fastq',\n",
       " 'sg_609.fastq',\n",
       " 'sg_610.fastq',\n",
       " 'sg_611.fastq',\n",
       " 'sg_612.fastq',\n",
       " 'sg_613.fastq',\n",
       " 'sg_614.fastq',\n",
       " 'sg_615.fastq',\n",
       " 'sg_616.fastq',\n",
       " 'sg_617.fastq',\n",
       " 'sg_618.fastq',\n",
       " 'sg_619.fastq',\n",
       " 'sg_620.fastq',\n",
       " 'sg_621.fastq',\n",
       " 'sg_622.fastq',\n",
       " 'sg_623.fastq',\n",
       " 'sg_624.fastq',\n",
       " 'sg_625.fastq',\n",
       " 'sg_626.fastq',\n",
       " 'sg_627.fastq',\n",
       " 'sg_628.fastq',\n",
       " 'sg_629.fastq',\n",
       " 'sg_630.fastq',\n",
       " 'sg_631.fastq',\n",
       " 'sg_632.fastq',\n",
       " 'sg_633.fastq',\n",
       " 'sg_634.fastq',\n",
       " 'sg_635.fastq',\n",
       " 'sg_636.fastq',\n",
       " 'sg_637.fastq',\n",
       " 'sg_638.fastq',\n",
       " 'sg_639.fastq',\n",
       " 'sg_640.fastq',\n",
       " 'sg_641.fastq',\n",
       " 'sg_642.fastq',\n",
       " 'sg_643.fastq',\n",
       " 'sg_644.fastq',\n",
       " 'sg_645.fastq',\n",
       " 'sg_646.fastq',\n",
       " 'sg_647.fastq',\n",
       " 'sg_648.fastq',\n",
       " 'sg_649.fastq',\n",
       " 'sg_650.fastq',\n",
       " 'sg_651.fastq',\n",
       " 'sg_652.fastq',\n",
       " 'sg_653.fastq',\n",
       " 'sg_654.fastq',\n",
       " 'sg_655.fastq',\n",
       " 'sg_656.fastq',\n",
       " 'sg_657.fastq',\n",
       " 'sg_658.fastq',\n",
       " 'sg_659.fastq',\n",
       " 'sg_660.fastq',\n",
       " 'sg_661.fastq',\n",
       " 'sg_662.fastq',\n",
       " 'sg_663.fastq',\n",
       " 'sg_664.fastq',\n",
       " 'sg_665.fastq',\n",
       " 'sg_666.fastq',\n",
       " 'sg_667.fastq',\n",
       " 'sg_668.fastq',\n",
       " 'sg_669.fastq',\n",
       " 'sg_670.fastq',\n",
       " 'sg_671.fastq',\n",
       " 'sg_672.fastq',\n",
       " 'sg_673.fastq',\n",
       " 'sg_674.fastq',\n",
       " 'sg_675.fastq',\n",
       " 'sg_676.fastq',\n",
       " 'sg_677.fastq',\n",
       " 'sg_678.fastq',\n",
       " 'sg_679.fastq',\n",
       " 'sg_680.fastq',\n",
       " 'sg_681.fastq',\n",
       " 'sg_682.fastq',\n",
       " 'sg_683.fastq',\n",
       " 'sg_684.fastq',\n",
       " 'sg_685.fastq',\n",
       " 'sg_686.fastq',\n",
       " 'sg_687.fastq',\n",
       " 'sg_688.fastq',\n",
       " 'sg_689.fastq',\n",
       " 'sg_690.fastq',\n",
       " 'sg_691.fastq',\n",
       " 'sg_692.fastq',\n",
       " 'sg_693.fastq',\n",
       " 'sg_694.fastq',\n",
       " 'sg_695.fastq',\n",
       " 'sg_696.fastq',\n",
       " 'sg_697.fastq',\n",
       " 'sg_698.fastq',\n",
       " 'sg_699.fastq',\n",
       " 'sg_700.fastq',\n",
       " 'sg_701.fastq',\n",
       " 'sg_702.fastq',\n",
       " 'sg_703.fastq',\n",
       " 'sg_704.fastq',\n",
       " 'sg_705.fastq',\n",
       " 'sg_706.fastq',\n",
       " 'sg_707.fastq',\n",
       " 'sg_708.fastq',\n",
       " 'sg_709.fastq',\n",
       " 'sg_710.fastq',\n",
       " 'sg_711.fastq',\n",
       " 'sg_712.fastq',\n",
       " 'sg_713.fastq',\n",
       " 'sg_714.fastq',\n",
       " 'sg_715.fastq',\n",
       " 'sg_716.fastq',\n",
       " 'sg_717.fastq',\n",
       " 'sg_718.fastq',\n",
       " 'sg_719.fastq',\n",
       " 'sg_720.fastq',\n",
       " 'sg_721.fastq',\n",
       " 'sg_722.fastq',\n",
       " 'sg_723.fastq',\n",
       " 'sg_724.fastq',\n",
       " 'sg_725.fastq',\n",
       " 'sg_726.fastq',\n",
       " 'sg_727.fastq',\n",
       " 'sg_728.fastq',\n",
       " 'sg_729.fastq',\n",
       " 'sg_730.fastq',\n",
       " 'sg_731.fastq',\n",
       " 'sg_732.fastq',\n",
       " 'sg_733.fastq',\n",
       " 'sg_734.fastq',\n",
       " 'sg_735.fastq',\n",
       " 'sg_736.fastq',\n",
       " 'sg_737.fastq',\n",
       " 'sg_738.fastq',\n",
       " 'sg_739.fastq',\n",
       " 'sg_740.fastq',\n",
       " 'sg_741.fastq',\n",
       " 'sg_742.fastq',\n",
       " 'sg_743.fastq',\n",
       " 'sg_744.fastq',\n",
       " 'sg_745.fastq',\n",
       " 'sg_746.fastq',\n",
       " 'sg_747.fastq',\n",
       " 'sg_748.fastq',\n",
       " 'sg_749.fastq',\n",
       " 'sg_750.fastq',\n",
       " 'sg_751.fastq',\n",
       " 'sg_752.fastq',\n",
       " 'sg_753.fastq',\n",
       " 'sg_754.fastq',\n",
       " 'sg_755.fastq',\n",
       " 'sg_756.fastq',\n",
       " 'sg_757.fastq',\n",
       " 'sg_758.fastq',\n",
       " 'sg_759.fastq',\n",
       " 'sg_760.fastq',\n",
       " 'sg_761.fastq',\n",
       " 'sg_762.fastq',\n",
       " 'sg_763.fastq',\n",
       " 'sg_764.fastq',\n",
       " 'sg_765.fastq',\n",
       " 'sg_766.fastq',\n",
       " 'sg_767.fastq',\n",
       " 'sg_768.fastq',\n",
       " 'sg_769.fastq',\n",
       " 'sg_770.fastq',\n",
       " 'sg_771.fastq',\n",
       " 'sg_772.fastq',\n",
       " 'sg_773.fastq',\n",
       " 'sg_774.fastq',\n",
       " 'sg_775.fastq',\n",
       " 'sg_776.fastq',\n",
       " 'sg_777.fastq',\n",
       " 'sg_778.fastq',\n",
       " 'sg_779.fastq',\n",
       " 'sg_780.fastq',\n",
       " 'sg_781.fastq',\n",
       " 'sg_782.fastq',\n",
       " 'sg_783.fastq',\n",
       " 'sg_784.fastq',\n",
       " 'sg_785.fastq',\n",
       " 'sg_786.fastq',\n",
       " 'sg_787.fastq',\n",
       " 'sg_788.fastq',\n",
       " 'sg_789.fastq',\n",
       " 'sg_790.fastq',\n",
       " 'sg_791.fastq',\n",
       " 'sg_792.fastq',\n",
       " 'sg_793.fastq',\n",
       " 'sg_794.fastq',\n",
       " 'sg_795.fastq',\n",
       " 'sg_796.fastq',\n",
       " 'sg_797.fastq',\n",
       " 'sg_798.fastq',\n",
       " 'sg_799.fastq',\n",
       " 'sg_800.fastq',\n",
       " 'sg_801.fastq',\n",
       " 'sg_802.fastq',\n",
       " 'sg_803.fastq',\n",
       " 'sg_804.fastq',\n",
       " 'sg_805.fastq',\n",
       " 'sg_806.fastq',\n",
       " 'sg_807.fastq',\n",
       " 'sg_808.fastq',\n",
       " 'sg_809.fastq',\n",
       " 'sg_810.fastq',\n",
       " 'sg_811.fastq',\n",
       " 'sg_812.fastq',\n",
       " 'sg_813.fastq',\n",
       " 'sg_814.fastq',\n",
       " 'sg_815.fastq',\n",
       " 'sg_816.fastq',\n",
       " 'sg_817.fastq',\n",
       " 'sg_818.fastq',\n",
       " 'sg_819.fastq',\n",
       " 'sg_820.fastq',\n",
       " 'sg_821.fastq',\n",
       " 'sg_822.fastq',\n",
       " 'sg_823.fastq',\n",
       " 'sg_824.fastq',\n",
       " 'sg_825.fastq',\n",
       " 'sg_826.fastq',\n",
       " 'sg_827.fastq',\n",
       " 'sg_828.fastq',\n",
       " 'sg_829.fastq',\n",
       " 'sg_830.fastq',\n",
       " 'sg_831.fastq',\n",
       " 'sg_832.fastq',\n",
       " 'sg_833.fastq',\n",
       " 'sg_834.fastq',\n",
       " 'sg_835.fastq',\n",
       " 'sg_836.fastq',\n",
       " 'sg_837.fastq',\n",
       " 'sg_838.fastq',\n",
       " 'sg_839.fastq',\n",
       " 'sg_840.fastq',\n",
       " 'sg_841.fastq',\n",
       " 'sg_842.fastq',\n",
       " 'sg_843.fastq',\n",
       " 'sg_844.fastq',\n",
       " 'sg_845.fastq',\n",
       " 'sg_846.fastq',\n",
       " 'sg_847.fastq',\n",
       " 'sg_848.fastq',\n",
       " 'sg_849.fastq',\n",
       " 'sg_850.fastq',\n",
       " 'sg_851.fastq',\n",
       " 'sg_852.fastq',\n",
       " 'sg_853.fastq',\n",
       " 'sg_854.fastq',\n",
       " 'sg_855.fastq',\n",
       " 'sg_856.fastq',\n",
       " 'sg_857.fastq',\n",
       " 'sg_858.fastq',\n",
       " 'sg_859.fastq',\n",
       " 'sg_860.fastq',\n",
       " 'sg_861.fastq',\n",
       " 'sg_862.fastq',\n",
       " 'sg_863.fastq',\n",
       " 'sg_864.fastq',\n",
       " 'sg_865.fastq',\n",
       " 'sg_866.fastq',\n",
       " 'sg_867.fastq',\n",
       " 'sg_868.fastq',\n",
       " 'sg_869.fastq',\n",
       " 'sg_870.fastq',\n",
       " 'sg_871.fastq',\n",
       " 'sg_872.fastq',\n",
       " 'sg_873.fastq',\n",
       " 'sg_874.fastq',\n",
       " 'sg_875.fastq',\n",
       " 'sg_876.fastq',\n",
       " 'sg_877.fastq',\n",
       " 'sg_878.fastq',\n",
       " 'sg_879.fastq',\n",
       " 'sg_880.fastq',\n",
       " 'sg_881.fastq',\n",
       " 'sg_882.fastq',\n",
       " 'sg_883.fastq',\n",
       " 'sg_884.fastq',\n",
       " 'sg_885.fastq',\n",
       " 'sg_886.fastq',\n",
       " 'sg_887.fastq',\n",
       " 'sg_888.fastq',\n",
       " 'sg_889.fastq',\n",
       " 'sg_890.fastq',\n",
       " 'sg_891.fastq',\n",
       " 'sg_892.fastq',\n",
       " 'sg_893.fastq',\n",
       " 'sg_894.fastq',\n",
       " 'sg_895.fastq',\n",
       " 'sg_896.fastq',\n",
       " 'sg_897.fastq',\n",
       " 'sg_898.fastq',\n",
       " 'sg_899.fastq',\n",
       " 'sg_900.fastq',\n",
       " 'sg_901.fastq',\n",
       " 'sg_902.fastq',\n",
       " 'sg_903.fastq',\n",
       " 'sg_904.fastq',\n",
       " 'sg_905.fastq',\n",
       " 'sg_906.fastq',\n",
       " 'sg_907.fastq',\n",
       " 'sg_908.fastq',\n",
       " 'sg_909.fastq',\n",
       " 'sg_910.fastq',\n",
       " 'sg_911.fastq',\n",
       " 'sg_912.fastq',\n",
       " 'sg_913.fastq',\n",
       " 'sg_914.fastq',\n",
       " 'sg_915.fastq',\n",
       " 'sg_916.fastq',\n",
       " 'sg_917.fastq',\n",
       " 'sg_918.fastq',\n",
       " 'sg_919.fastq',\n",
       " 'sg_920.fastq',\n",
       " 'sg_921.fastq',\n",
       " 'sg_922.fastq',\n",
       " 'sg_923.fastq',\n",
       " 'sg_924.fastq',\n",
       " 'sg_925.fastq',\n",
       " 'sg_926.fastq',\n",
       " 'sg_927.fastq',\n",
       " 'sg_928.fastq',\n",
       " 'sg_929.fastq',\n",
       " 'sg_930.fastq',\n",
       " 'sg_931.fastq',\n",
       " 'sg_932.fastq',\n",
       " 'sg_933.fastq',\n",
       " 'sg_934.fastq',\n",
       " 'sg_935.fastq',\n",
       " 'sg_936.fastq',\n",
       " 'sg_937.fastq',\n",
       " 'sg_938.fastq',\n",
       " 'sg_939.fastq',\n",
       " 'sg_940.fastq',\n",
       " 'sg_941.fastq',\n",
       " 'sg_942.fastq',\n",
       " 'sg_943.fastq',\n",
       " 'sg_944.fastq',\n",
       " 'sg_945.fastq',\n",
       " 'sg_946.fastq',\n",
       " 'sg_947.fastq',\n",
       " 'sg_948.fastq',\n",
       " 'sg_949.fastq',\n",
       " 'sg_950.fastq',\n",
       " 'sg_951.fastq',\n",
       " 'sg_952.fastq',\n",
       " 'sg_953.fastq',\n",
       " 'sg_954.fastq',\n",
       " 'sg_955.fastq',\n",
       " 'sg_956.fastq',\n",
       " 'sg_957.fastq',\n",
       " 'sg_958.fastq',\n",
       " 'sg_959.fastq',\n",
       " 'sg_960.fastq',\n",
       " 'sg_961.fastq',\n",
       " 'sg_962.fastq',\n",
       " 'sg_963.fastq',\n",
       " 'sg_964.fastq',\n",
       " 'sg_965.fastq',\n",
       " 'sg_966.fastq',\n",
       " 'sg_967.fastq',\n",
       " 'sg_968.fastq',\n",
       " 'sg_969.fastq',\n",
       " 'sg_970.fastq',\n",
       " 'sg_971.fastq',\n",
       " 'sg_972.fastq',\n",
       " 'sg_973.fastq',\n",
       " 'sg_974.fastq',\n",
       " 'sg_975.fastq',\n",
       " 'sg_976.fastq',\n",
       " 'sg_977.fastq',\n",
       " 'sg_978.fastq',\n",
       " 'sg_979.fastq',\n",
       " 'sg_980.fastq',\n",
       " 'sg_981.fastq',\n",
       " 'sg_982.fastq',\n",
       " 'sg_983.fastq',\n",
       " 'sg_984.fastq',\n",
       " 'sg_985.fastq',\n",
       " 'sg_986.fastq',\n",
       " 'sg_987.fastq',\n",
       " 'sg_988.fastq',\n",
       " 'sg_989.fastq',\n",
       " 'sg_990.fastq',\n",
       " 'sg_991.fastq',\n",
       " 'sg_992.fastq',\n",
       " 'sg_993.fastq',\n",
       " 'sg_994.fastq',\n",
       " 'sg_995.fastq',\n",
       " 'sg_996.fastq',\n",
       " 'sg_997.fastq',\n",
       " 'sg_998.fastq',\n",
       " 'sg_999.fastq',\n",
       " 'sg_1000.fastq',\n",
       " ...]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir('/Volumes/sanchezrivera/samgould/240624San/D24-10888-6775T_guide_split')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df2.to_csv('be_cluster_scripts_2024_7_1/IDR_ABE_library_columns_renamed.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example post-processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def crispresso_compiler(samp, sample_id):\n",
    "    \"\"\" \n",
    "    Takes in list of sample names (see cell above)\n",
    "    Returns compiled dictionary of dataframes containing editing information for each sensor\n",
    "    - key for each is the sample_id name\n",
    "    \n",
    "    Includes:\n",
    "    1. corr_perc = pure correct editing\n",
    "    2. target_base_edit_perc = target base editing perc (including edits with bystander editing)\n",
    "    3. wt_perc\n",
    "    4. byproduct information (indels, substitutions, ambiguous)\n",
    "    \"\"\"\n",
    "\n",
    "    df_edits = []\n",
    "    for k in samp: \n",
    "        concated = pd.read_csv(f'crispresso/{k}_crispresso_aggregated.csv')\n",
    "        concated = concated.fillna(0)\n",
    "\n",
    "        #somehow wasn't able to find the initial set of code that I used for this...\n",
    "\n",
    "        #go through all of the samples and do it step by step\n",
    "        sample_ids = []\n",
    "        #sample_num = []\n",
    "        rii = []\n",
    "        raaa = []\n",
    "        r_lowqual = []\n",
    "        ra_hdr = []\n",
    "        ra_wt = []\n",
    "        r_unaligned = []\n",
    "        no_edit = []\n",
    "        correct = []\n",
    "        target_base_editing = []\n",
    "        byproduct_all = []\n",
    "        byproduct_indel = []\n",
    "        byproduct_sub = []\n",
    "        byproduct_ambig = []\n",
    "        for i in np.unique(concated['Guide_ID']):\n",
    "            sample_ids.append(i)\n",
    "\n",
    "            subset = concated[concated['Guide_ID']==i]\n",
    "\n",
    "            wt = subset[subset['Amplicon']=='Reference']\n",
    "            edit = subset[subset['Amplicon']=='HDR']\n",
    "\n",
    "            #sample_num.append(wt['sample_num'].values[0])\n",
    "            rii1 = wt['Reads_in_input'].values[0]\n",
    "            r1 = wt['Reads_aligned_all_amplicons'].values[0]\n",
    "            rii.append(rii1)\n",
    "            raaa.append(r1)\n",
    "            r_lowqual.append(rii1-r1)\n",
    "\n",
    "            r2 = edit['Reads_aligned'].values[0]\n",
    "            r3 = wt['Reads_aligned'].values[0]\n",
    "            ra_hdr.append(r2)\n",
    "            ra_wt.append(r3)\n",
    "            r_unaligned.append(r1 - (r2+r3))\n",
    "\n",
    "            no_edit.append(wt['Unmodified'].values[0])\n",
    "            correct.append(edit['Unmodified'].values[0])\n",
    "            target_base_editing.append(edit['Modified'].values[0] + edit['Unmodified'].values[0])\n",
    "\n",
    "            byprod_all = wt['Modified'].values[0] + edit['Modified'].values[0] + (r1 - (r2+r3)) #add unaligned reads\n",
    "            sub_all = wt['Only Substitutions'].values[0] + edit['Only Substitutions'].values[0]\n",
    "            indel_all = wt['Only Deletions'].values[0] + wt['Only Insertions'].values[0] + wt['Insertions and Deletions'].values[0] + edit['Only Deletions'].values[0] + edit['Only Insertions'].values[0] + edit['Insertions and Deletions'].values[0]\n",
    "            ambig_all = byprod_all - sub_all - indel_all\n",
    "\n",
    "            byproduct_all.append(byprod_all)\n",
    "            byproduct_indel.append(indel_all)\n",
    "            byproduct_sub.append(sub_all)\n",
    "            byproduct_ambig.append(ambig_all)\n",
    "\n",
    "\n",
    "        cols = [\"Guide_ID\", \"Reads_in_input\",\"Reads_lowqual\", \"Reads_aligned_all_amplicons\",\"Reads_aligned_WT\", \"Reads_aligned_HDR\", \"Reads_unaligned\", \"WT\",\"correct_edit\", \"target_base_edit\", \"byproduct_all\",\"byproduct_INDEL\",\"byproduct_sub\",\"byproduct_ambiguous\"]\n",
    "        col_vals = [ sample_ids, rii, r_lowqual, raaa, ra_wt, ra_hdr, r_unaligned, no_edit, correct, target_base_editing, byproduct_all, byproduct_indel, byproduct_sub, byproduct_ambig]\n",
    "\n",
    "        out = pd.DataFrame(dict(zip(cols, col_vals)))\n",
    "        out['corr_perc'] = 100*(out['correct_edit']/out['Reads_aligned_all_amplicons'])\n",
    "        out['target_base_edit_perc'] = 100*(out['target_base_edit']/out['Reads_aligned_all_amplicons'])\n",
    "        out['WT_perc'] = 100*(out['WT']/out['Reads_aligned_all_amplicons'])\n",
    "        out['byproduct_all_perc'] =  100*(out['byproduct_all']/out['Reads_aligned_all_amplicons'])\n",
    "        out['byproduct_INDEL_perc'] =  100*(out['byproduct_INDEL']/out['Reads_aligned_all_amplicons'])\n",
    "        out['byproduct_sub_perc'] =  100*(out['byproduct_sub']/out['Reads_aligned_all_amplicons'])\n",
    "        out['byproduct_ambiguous_perc'] =100*(out['byproduct_ambiguous']/out['Reads_aligned_all_amplicons'])\n",
    "        out = out.fillna(0)\n",
    "\n",
    "        df_edits.append(out)\n",
    "\n",
    "    edit_dict = dict(zip(sample_id, df_edits))\n",
    "\n",
    "    return edit_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#example_inputs\n",
    "samp = ['D24-10888-6775T_guide_split',\n",
    " 'D24-10889-6775T_guide_split',\n",
    " 'D24-10890-6775T_guide_split',\n",
    " 'D24-10891-6775T_guide_split',\n",
    " 'D24-10892-6775T_guide_split',\n",
    " 'D24-10893-6775T_guide_split',]\n",
    "\n",
    "sample_id = ['sample1', 'sample2', 'sample3', 'sample4', 'sample5', 'sample5']\n",
    "\n",
    "edit_dict = crispresso_compiler(samp, sample_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#can then access editing through \n",
    "edit_dict['sample1']"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
